rfcs/debuggability_metrics.md

```diff
@@ -0,0 +1,126 @@
+
+### Implementation
+
+* Write script that scrapes Jenkins build logs to measure Failure first signaled and Failure e-mail sent metrics
+* Perform external developer polling (this would be where most questions would be) (this would include log-finding, log-parsing, artifact-finding metrics)
+
+### Possible Outcomes / Scenarios
+
+Between metrics and polling, we will be able to generate a better picture of our pain sources and problems than we have now. While we won't be able to predict every possible outcome of these metrics and polling, we will attempt to identify several likely ones and comment briefly on our plans for those outcomes.
+
+Outcome/Scenario | Planned Steps
+-- | --
+Problems are not being logged due to unclear issue-filing process | Provide/improve documentation on issue-filing, run trainings on issue-filing
+Problems are not being logged due to effort required | Provide quick-links and other shortcuts, create issue templates
+Problems are being logged, but are not being actioned on | Use metrics to provide data to prioritize issues higher, reach out to appropriate owners
+
+## What are the advantages of the suggestion?
+
+* Provides absolute metrics in debugability (see table above)
+* Polls customers directly to get their experiences
+* Focuses on the key aspects of debugability and provides a limited scope
+* Low investment cost
+
+## What are the disadvantages of the suggestion?
+
+* Not all aspects are measured with metrics, meaning that some pain points may be under-examined
+* Metrics are only a partial picture of the problem, potentially supporting some solutions more than they merit (this can be mitigated by seeking customer feedback)
+* Customer opinions may change over time, requiring multiple polls (may require setting a particular cadence in order to maximize responses)
+* Some issues found by metrics/polling may not be relevant to the scope
+
+## Are there any alternatives to this suggestion?
+
+This suggestion has multiple components that are not coupled, so each component can be considered individually. This runs the risk of giving a partial picture, but perhaps the lesser amount of investment is worth the lesser result.
+
+We could implement metrics on every step of the debugging process. This has a high risk of noise, as certain steps running long are way less impactful than others (builds can be run overnight, for example, while finding the logs and artifacts is an active step). Further, many of these metrics can be addressed by tasks out of scope for this RFC.
+
+We could skip the detailed customer experience poll. Broad polling often has low engagement, however, so a more direct interaction is expected to provide better results, even if it is at a higher cost.
+
+## What is the strategy for adoption?
```

Comment:
Perhaps a better strategy for debugging pain points is to encourage cutting issues for inefficient workflows? This would crowdsource the UX information, though not be standardized.